host/xtest/regression_1000.c

```diff
@@ -1715,3 +1715,82 @@ static void xtest_tee_test_1021(ADBG_Case_t *c)
 }
 ADBG_CASE_DEFINE(regression, 1021, xtest_tee_test_1021,
 		 "Test panic context release");
+
+static void xtest_tee_test_1022(ADBG_Case_t *c)
+{
+	TEEC_Session session = { 0 };
+	TEEC_Operation op = TEEC_OPERATION_INITIALIZER;
+	uint32_t ret_orig;
+	uint8_t *empty_buf = NULL;
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+		xtest_teec_open_session(&session, &os_test_ta_uuid, NULL,
+			&ret_orig)))
+		goto out;
+
+	empty_buf = malloc(0);
+	if (!empty_buf) {
+		(void)ADBG_EXPECT_TEEC_SUCCESS(c, TEEC_ERROR_OUT_OF_MEMORY);
+		goto out;
+	}
+
+	Do_ADBG_BeginSubCase(c,
+			"Input/Output MEMREF Buffer NULL - Size 0 bytes");
+
+	op.paramTypes = TEEC_PARAM_TYPES(TEEC_MEMREF_TEMP_INPUT,
+					 TEEC_MEMREF_TEMP_INPUT,
+					 TEEC_MEMREF_TEMP_OUTPUT,
+					 TEEC_MEMREF_TEMP_OUTPUT);
+
+	/*
+	 * Check temp buffer parameters with a valid address
+	 * but 0 bytes length
+	 */
+	op.params[0].tmpref.buffer = empty_buf;
+	op.params[0].tmpref.size = 0;
+
+	op.params[1].tmpref.buffer = NULL;
+	op.params[1].tmpref.size = 0;
+
+	op.params[2].tmpref.buffer = empty_buf;
+	op.params[2].tmpref.size = 0;
+
+	op.params[3].tmpref.buffer = NULL;
+	op.params[3].tmpref.size = 0;
+
+	ADBG_EXPECT(c, TEE_SUCCESS,
+			TEEC_InvokeCommand(&session,
+				TA_OS_TEST_CMD_PARAMS_TMPMEM, &op, &ret_orig));
+
+	Do_ADBG_EndSubCase(c, NULL);
```

Comment:
prefer "Input/Output MEMREF Buffer NULL - Size 0 bytes" over NULL for closing.
Here and below.

Reply:
Test and description are correct. The "Input/Ouput MEMREF Buffer NULL - Size 0 bytes" is done before. Here is to test the correct behaviour when Input MEMREF buffer is NULL pointer but size non 0 bytes:

Reply:
@cneveux , I think @etienne-lms  refers to just have the "group" confined, i.e., same message 
as in "`Do_ADBG_BeginSubCase`", correct @etienne-lms ?

Reply:
true.

```C
	/*
	 * Check temp buffer parameters with a valid address
	 * but 0 bytes length
	 */
	Do_ADBG_BeginSubCase(c,
			     "Input/Output MEMREF Buffer NULL - Size 0 bytes");

	(...)

	Do_ADBG_EndSubCase(c, "Input/Output MEMREF Buffer NULL - Size 0 bytes");


	/*
	 * Check temp buffer parameters with a valid address
	 * but non 0 bytes length
	 */
	Do_ADBG_BeginSubCase(c, "Input MEMREF Buffer NULL - Size non 0 bytes");

	(...)

	Do_ADBG_EndSubCase(c, "Input MEMREF Buffer NULL - Size non 0 bytes");
```

Note that inline comment are redundant with log message, could remove the inline comment.

Reply:
ok